tiledb/api/src/query/write/input.rs

```diff
@@ -5,6 +5,24 @@ pub trait DataProvider {
     fn as_tiledb_input(&self) -> QueryBuffers;
 }
 
+impl<'data, T> DataProvider for QueryBuffers<'data, T> {
```

Comment:
I'm pretty sure I tried adding `QueryBuffersMut` at the same time thinking the same thing but something about the borrow or lifetimes confused me for long enough I just deleted it rather than get side tracked by the issue. Might be one of those "obvious in hindsight" things if I try it again so I'll do that.